src/components/selection.ts

```diff
@@ -353,24 +353,47 @@ export default class SelectionUtils {
   }
 
   /**
-   * Adds fake cursor to the current range
+   * Check if current range exists and belongs to container
    *
-   * @param [container] - if passed cursor will be added only if container contains current range
+   * @param container - where range should be
    */
-  public static addFakeCursor(container?: HTMLElement): void {
+  public static isRangeInsideContainer(container: HTMLElement): boolean {
     const range = SelectionUtils.range;
-    const fakeCursor = $.make('span', 'codex-editor__fake-cursor');
 
-    fakeCursor.dataset.mutationFree = 'true';
+    if (range === null) {
+      return false;
+    }
+
+    return container.contains(range.startContainer);
```

Comment:
Shouldn't you also check `endContainer`?

Reply:
I think it's enough to know that a container has a range starting node. And this statement was used previously inside `addFakeCursor`, I've just moved it to a separate method